src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs

```diff
@@ -349,13 +376,88 @@ private bool IsFileSource()
 
                         return Task.FromResult(0);
                     },
                     logger,
                     token);
             }
+
+            return showPushCommandPackagePushed;
+        }
+
+        /// <summary>
+        /// Gently log any specified ContinueOnError status code without throwing.
+        /// </summary>
+        /// <param name="continuedErrorStatusCode">If provided, it indicates that this StatusCode occurred but was flagged as a ContinueOn.</param>
+        /// <param name="logger"></param>
+        /// <returns>Indication of whether the log occurred.</returns>
+        private static bool DetectAndLogContinueOnErrorOccurrence(HttpStatusCode? continuedErrorStatusCode, ILogger logger)
+        {
+            bool continueOnErrorOccurred = false;
+
+            if (continuedErrorStatusCode != null)
+            {
+                string messageToLog = null;
+                switch (continuedErrorStatusCode.Value)
+                {
+                    case HttpStatusCode.Conflict:
+                        messageToLog = Strings.PushCommandSkipDuplicate;
+                        continueOnErrorOccurred = true;
+                        break;
+                    case HttpStatusCode.BadRequest:
+                        messageToLog = Strings.PushCommandSkipInvalid;
+                        continueOnErrorOccurred = true;
+                        break;
+                    default: break; //Not supported ContinueOnError code.
+                }
+                if (messageToLog != null)
+                {
+                    logger?.LogInformation(messageToLog);
+                }
+            }
+
+            return continueOnErrorOccurred;
+        }
+
+        /// <summary>
+        /// Ensures a Success HTTP Status Code is returned unless a specified exclusion occurred. If CodesNotToThrow is provided and the response contains
+        /// one of these codes, do not EnsureSuccess and instead return the exception code gracefully.
+        /// </summary>
+        /// <param name="response"></param>
+        /// <param name="codesNotToThrow">List of exclusions that will not cause a thrown Exception.</param>
+        /// <param name="continuedErrorPossibleOptions">All options that ContinueOnError can be applied to.</param>
+        /// <param name="logger"></param>
+        /// <returns>Response StatusCode</returns>
+        private static HttpStatusCode? EnsureSuccessStatusCode(HttpResponseMessage response, List<HttpStatusCode> codesNotToThrow,
+                                                               List<HttpStatusCode> continuedErrorPossibleOptions, ILogger logger)
+        {
+            //If this status code is to be excluded.
+            if (codesNotToThrow != null && codesNotToThrow.Contains(response.StatusCode))
+            {
+                return response.StatusCode;
+            }
+            else if (continuedErrorPossibleOptions.Contains(response.StatusCode))
```

Comment:
Probably won't use a Dictionary since actually we are now going to reduce this to 1. Don't expect the list to grow any, and certainly not substantially.